### PR TITLE
deepin.deepin-wm: fix paths related to plugins and desktop files

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- ~~deepin.deepin-wm: set plugins dir from environment variable~~
- deepin.deepin-wm: use absolute path in desktop file

About packaging deepin for NixOS: https://github.com/NixOS/nixpkgs/issues/59023

See also https://github.com/NixOS/nixpkgs/pull/59244#issuecomment-485097016

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+--- a/src/PluginManager.vala
++++ b/src/PluginManager.vala
+@@ -58,7 +58,10 @@ namespace Gala
+ 				return;
+ 			}
+ 
+-			plugin_dir = File.new_for_path (Config.PLUGINDIR);
++                        string? plugin_dir_name = GLib.Environment.get_variable ("DEEPIN_WM_PLUGINS_DIR");
++                        if (plugin_dir_name == null || plugin_dir_name == "")
++                        	plugin_dir_name = Config.PLUGINDIR;
```
Comment (worldofpeace) [inline pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch]: Can you add `debug ();` messages?
So is `plugin_dir` set and its value or it's not and the value of `Config.PLUGINDIR`?

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Are there any `deepin-wm` plugins distributed or used internally in other DDE components?
This will also need a wrapper used when there's plugins provided.

**Comment by romildo** [comment]
> Are there any `deepin-wm` plugins distributed or used internally in other DDE components?

I did not find any.

File: pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch]: > So is `plugin_dir` set and its value or it's not and the value of `Config.PLUGINDIR`?

Both are accepted, and the environment variable has priority.

Comment (romildo) [reply pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch]: > Can you add `debug ();` messages?

Done.

Comment (worldofpeace) [reply pkgs/desktops/deepin/deepin-wm/deepin-wm.plugins-dir.patch]: > Both are accepted, and the environment variable has priority.

Oh no, that's what I meant for the debug message to get across.

**Comment by romildo** [comment]
Considering that there is no other DDE component that installs any plugins for `deepin-wm`, it may be more convenient now to not patch the source code to set plugins dir from environment variable. Also the wrapper is not needed currently.

@worldofpeace Do you agree?

**Comment by worldofpeace** [comment]
Yeah, I think there's a comment somewhere from me about patching only when there's a plugin ecosystem that's useful. For example in pantheon `gala` isn't patched because almost all development work for plugins eventually get upstreamed. And from the looks of it this is a fork of `gala`.

**Comment by worldofpeace** [comment]
##### Though I wouldn't toss the code away, it could change in the future who knows

**Comment by romildo** [comment]
I am removing the `deepin-wm.plugins-dir.patch` from this PR, and keeping its contents in this comment, case it is needed later on.

``` patch
From 028f9214a251741df89b56836fa4e8efd16c8cb2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Romildo=20Malaquias?= <malaquias@gmail.com>
Date: Fri, 19 Apr 2019 23:56:33 -0300
Subject: [PATCH] deepin.deepin-wm: set plugin dir from environment variable

---
 src/PluginManager.vala | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/PluginManager.vala b/src/PluginManager.vala
index 9ff8856..f9ac214 100644
--- a/src/PluginManager.vala
+++ b/src/PluginManager.vala
@@ -58,7 +58,11 @@ namespace Gala
 				return;
 			}
 
-			plugin_dir = File.new_for_path (Config.PLUGINDIR);
+                        string? plugin_dir_name = GLib.Environment.get_variable ("DEEPIN_WM_PLUGINS_DIR");
+                        if (plugin_dir_name == null || plugin_dir_name == "")
+                        	plugin_dir_name = Config.PLUGINDIR;
+			debug ("Plugin dir: %s", plugin_dir_name);
+			plugin_dir = File.new_for_path (plugin_dir_name);
 			if (!plugin_dir.query_exists ())
 				return;
 
-- 
2.21.0
```

**Pull request merged**